fl4health/utils/metrics.py

```diff
@@ -133,140 +260,51 @@ def __call__(self, logits: torch.Tensor, target: torch.Tensor) -> Scalar:
-        for metric in self.metrics:
-            metric_values.append(metric(stacked_inputs, stacked_targets))
-
-        results: Dict[str, Scalar] = {
-            f"{self.name}_{str(metric)}".lstrip("_"): metric_value
-            for metric, metric_value in zip(self.metrics, metric_values)
-        }
-
-        return results
-
-    def clear(self) -> None:
-        self.accumulated_inputs = []
-        self.accumulated_targets = []
-
-
-class MetricAverageMeter(MetricMeter):
-    """
-    class used to compute the average of metrics iteratively evaluated over a set of prediction-target pairings.
-    The constructor takes a list of type Metric. These metrics are then evaluated each time the update method is
-    called with predcitions and ground truth labels. The count corresponding to each evaluation is stored to ensure
-    the metrics average is accurate. The compute method is used to return a dictionairy of metrics along with their
-    current values.
-    """
-
-    def __init__(self, metrics: Sequence[Metric], name: str = "") -> None:
-        super().__init__(metrics, name)
-        self.metric_values_history: List[List[Scalar]] = [[] for _ in range(len(self.metrics))]
-        self.counts: List[int] = []
-
-    def update(self, input: torch.Tensor, target: torch.Tensor) -> None:
+class MetricManager:
+    def __init__(self, metrics: Sequence[Metric], metric_mngr_name: str) -> None:
         """
-        Evaluate metrics and store results.
-        """
-        metric_values: List[Scalar] = [metric(input, target) for metric in self.metrics]
-        self.counts.append(target.size(0))
-
-        for i, metric_value in enumerate(metric_values):
-            self.metric_values_history[i].append(metric_value)
+        Class to manage a set of metrics associated to a given prediction type.
 
-    def compute(self) -> Dict[str, Scalar]:
-        """
-        Returns average of each metrics given its historical values and counts
+        Args:
+            metrics (Sequence[Metric]): List of metric to evaluate predictions on.
+            metric_mngr_name (str): Name of the metric manager (ie train, val, test)
         """
-        total_count = sum(self.counts)
-        weights: List[float] = [count / total_count for count in self.counts]
-
-        metric_value_averages = []
-        for metric_values in self.metric_values_history:
-            avg = sum([weight * float(val) for weight, val in zip(weights, metric_values)])
-            metric_value_averages.append(avg)
-
-        results: Dict[str, Scalar] = {
-            f"{self.name}_{str(metric)}".lstrip("_"): avg for metric, avg in zip(self.metrics, metric_value_averages)
-        }
-
-        return results
-
-    def clear(self) -> None:
-        self.metric_values_history = [[] for _ in range(len(self.metrics))]
-        self.counts = []
+        self.og_metrics = metrics
+        self.metric_mngr_name = metric_mngr_name
```

Comment:
any objection to expanding these to `metric_manager_name ` and `metric_manager_name `, respectively since we're only saving three letters in the abbreviation anyway?

Reply:
There are a few other places in the code where we use this abbreviation that I'd suggest we expand as well, unless you really don't like it 🙂